main.go

```diff
@@ -55,175 +48,157 @@ func main() {
-			PeriodSeconds:       int32(cfg.LivenessProbePeriodSeconds),
+			InitialDelaySeconds: int32(config.LivenessProbeInitialDelaySeconds),
+			TimeoutSeconds:      int32(config.LivenessProbeTimeoutSeconds),
+			PeriodSeconds:       int32(config.LivenessProbePeriodSeconds),
 		},
-		ImagePullPolicy: cfg.ImagePullPolicy,
+		ImagePullPolicy:   config.ImagePullPolicy,
+		ProfilesNamespace: config.ProfilesNamespace,
 	}
 
-	factory := k8s.NewFunctionFactory(clientset, deployConfig)
+	log.Printf("DeploymentConfig: %+v\n", deployConfig)
 
-	defaultResync := time.Second * 5
-	kubeInformerOpt := kubeinformers.WithNamespace(functionNamespace)
-	kubeInformerFactory := kubeinformers.NewSharedInformerFactoryWithOptions(clientset, defaultResync, kubeInformerOpt)
+	// the sync interval does not affect the scale to/from zero feature
+	// auto-scaling is does via the HTTP API that acts on the deployment Spec.Replicas
+	defaultResync := time.Minute * 5
+
+	kubeInformerOpt := kubeinformers.WithNamespace(config.DefaultFunctionNamespace)
+	kubeInformerFactory := kubeinformers.NewSharedInformerFactoryWithOptions(kubeClient, defaultResync, kubeInformerOpt)
+
+	faasInformerOpt := informers.WithNamespace(config.DefaultFunctionNamespace)
+	faasInformerFactory := informers.NewSharedInformerFactoryWithOptions(faasClient, defaultResync, faasInformerOpt)
+
+	// this is where we need to swap to the faasInformerFactory
+	profileInformerOpt := informers.WithNamespace(config.ProfilesNamespace)
+	profileInformerFactory := informers.NewSharedInformerFactoryWithOptions(faasClient, defaultResync, profileInformerOpt)
+	profileLister := profileInformerFactory.Openfaas().V1().Profiles().Lister()
+	factory := k8s.NewFunctionFactory(kubeClient, deployConfig, profileLister)
+
+	setup := serverSetup{
+		config:                 config,
+		functionFactory:        factory,
+		kubeInformerFactory:    kubeInformerFactory,
+		faasInformerFactory:    faasInformerFactory,
+		profileInformerFactory: profileInformerFactory,
+		kubeClient:             kubeClient,
+		faasClient:             faasClient,
+	}
+
+	if !operator {
+		log.Println("Starting controller")
+		runController(setup)
+	} else {
+		log.Println("Starting operator")
+		runOperator(setup)
+	}
+}
+
+// runController runs the faas-netes imperative controller
+func runController(setup serverSetup) {
+	// pull out the required config and clients fromthe setup, this is largely a
+	// leftover from refactoring the setup to a shared step and keeping the function
+	// signature readable
+	config := setup.config
+	kubeClient := setup.kubeClient
+	factory := setup.functionFactory
+	kubeInformerFactory := setup.kubeInformerFactory
+	faasInformerFactory := setup.faasInformerFactory
 
 	// set up signals so we handle the first shutdown signal gracefully
 	stopCh := signals.SetupSignalHandler()
 
 	endpointsInformer := kubeInformerFactory.Core().V1().Endpoints()
+
+	log.Println("waiting for openfaas CRD cache sync")
```

Comment:
I think we usually capitalize the first line of log statements, but keep errors lower.